voice/src/main/kotlin/udp/RTPPacket.kt

```diff
@@ -38,8 +41,8 @@ public data class RTPPacket(
     public companion object {
         internal const val VERSION = 2
 
-        public fun fromPacket(packet: ByteReadPacket): RTPPacket? = with(packet) base@{
-            if (remaining <= 13) return@base null
+        public fun fromPacket(packet: Source): RTPPacket? = with(packet) base@{
+            if (!request(byteCount = 14)) return@base null
```

Comment:
@lost-illusi0n was the `<=` here and below intended? The RTP header is 12 bytes and then you required to have 2 more bytes.

Reply:
I believe I intentionally required it to have at least one more byte (12 byte header + a one byte payload), but wrote it like that. Woops. This could probably be changed, but a one byte payload is unlikely in reality (luckily). To be truly correct, an RTP packet could just be a header, with no payload. Maybe supporting that wouldn't be a bad idea. I just didn't bother (or forgot to) three years ago.

For [the second `<=` case](https://github.com/kordlib/kord/blame/3ef564ccdaa1659ff04305d76ee33756e47cd1e0/voice/src/main/kotlin/udp/RTPPacket.kt#L80), that was to keep the intentional requirement of a payload in place. Made the same mistake of requiring a two byte payload again though. At least it's consistent)) If the requirement of a payload is dropped, then these extra data requirements can also be dropped. If not, then changing the upper bound of the data checks to one less should be fine too.

Reply:
Alright, thanks for taking a look at this after 3 years :D

I think I'm going to leave the accidental +2 byte requirement in place for now. This PR is supposed to be a dependency upgrade and as you said it's unlikely that this causes problems in practice.

I'm [working](https://github.com/lukellmann/kord/tree/new-voice-encryption) on the [new encryption modes](https://discord.com/developers/docs/change-log#voice-encryption-modes) at the moment and I'm considering to touch the RTP logic with that too, so I'd rather bring in the change there.